src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/Contract.cs

```diff
@@ -74,11 +113,11 @@ public static void ThrowIfFalse([DoesNotReturnIf(parameterValue: false)] bool co
         /// in all builds
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void ThrowIfFalse([DoesNotReturnIf(parameterValue: false)] bool condition, string message, [CallerLineNumber] int lineNumber = 0)
+        public static void ThrowIfFalse([DoesNotReturnIf(parameterValue: false)] bool condition, [InterpolatedStringHandlerArgument("condition")] ThrowIfFalseInterpolatedStringHandler message, [CallerLineNumber] int lineNumber = 0)
```

Comment:
We should add these as overloads, not replacements. Overload resolution will pick the handler for interpolated constants.